### PR TITLE
improve mandatory field visibility

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: src/app/si/settings/connect-sage-intacct/connect-sage-intacct.ts [hunk 1]
```diff
@@ -32,29 +32,26 @@ export class ConnectSageIntacctComponent implements OnInit {
...
+    const connectSageIntacct = [
+      that.settingsService.connectSageIntacct(that.workspaceId, {
+        si_user_id: userID,
+        si_company_id: companyID,
+        si_company_name: companyName,
+        si_user_password: userPassword
+      })
+    ];
+
+    forkJoin(connectSageIntacct).subscribe(responses => {
```
Comment (ashwin1111) [inline src/app/si/settings/connect-sage-intacct/connect-sage-intacct.ts:46]: single api call doesn't need to be a fork join, please change this

File: src/app/si/settings/general-mappings/general-mappings.component.html [hunk 2]
```diff
@@ -115,10 +118,12 @@
           </mat-option>
         </mat-select>
       </mat-form-field>
+      <mandatory-error-message listName="default credit card vendor" *ngIf="form.controls.defaultVendor.touched && !form.controls.defaultVendor.valid"></mandatory-error-message>
       
       <div class="mat-body-strong" *ngIf="generalSettings.import_projects">
         <strong class="mappings-dialog--label">
           Default Item
+          <mandatory-field *ngIf="isFieldMandatory('defaultVendor')"></mandatory-field>
```
Comment (ashwin1111) [inline src/app/si/settings/general-mappings/general-mappings.component.html:126]: `isFieldMandatory('defaultItem') `
Please be very careful on these sections, single name change can be very dangerous

File: src/app/si/settings/general-mappings/general-mappings.component.html [hunk 3]
```diff
@@ -115,10 +118,12 @@
           </mat-option>
         </mat-select>
       </mat-form-field>
+      <mandatory-error-message listName="default credit card vendor" *ngIf="form.controls.defaultVendor.touched && !form.controls.defaultVendor.valid"></mandatory-error-message>
```
Comment (ashwin1111) [inline src/app/si/settings/general-mappings/general-mappings.component.html:121]: `listName` is incorrect, Please be very careful on these sections, single name change can be very dangerous

**Review by ashwin1111 — CHANGES_REQUESTED** [review]
@NileshPant1999 please do one round of self review and assign it back, have left few comments.

File: src/app/si/settings/general-mappings/general-mappings.component.ts [hunk 4]
```diff
@@ -47,60 +47,97 @@ export class GeneralMappingsComponent implements OnInit {
...
+      if (that.generalSettings.import_projects) {
+        that.form.controls.defaultItem.setValidators(Validators.required);
+      }
+
+      if (that.generalSettings.sync_fyle_to_sage_intacct_payments) {
+        that.form.controls.paymentAccount.setValidators(Validators.required);
+      }
+
+      if (that.generalSettings.reimbursable_expenses_object === 'EXPENSE_REPORT') {
+        that.form.controls.defaultReimbursableExpensePaymentType.setValidators(Validators.required);
```
Comment (ashwin1111) [inline src/app/si/settings/general-mappings/general-mappings.component.ts:122]: I don't find this in API

File: src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts [hunk 5]
```diff
@@ -175,82 +174,78 @@ export class ConfigurationComponent implements OnInit {
...
-          setTimeout(() => {
-            that.snackBar.open('Auto mapping of employees may take up to 10 minutes');
-          }, 1500);
-        }
+      that.si.getGeneralSettings();
 
-        that.router.navigateByUrl(`workspaces/${that.workspaceId}/dashboard`);
+      if (autoMapEmployees) {
+        setTimeout(() => {
+          that.snackBar.open('Auto mapping of employees may take up to 10 minutes');
```
Comment (ashwin1111) [inline src/app/si/settings/sage-intacct-configurations/configuration/configuration.component.ts:241]: `Auto mapping of employees may take few minutes`

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: src/app/si/settings/general-mappings/general-mappings.component.ts (hunk 4 above)
Comment (NileshPant1999) [reply src/app/si/settings/general-mappings/general-mappings.component.ts:122]: this is for sage_intacct reimbursement creation you can find it inside mappings/utils.py

File: src/app/si/settings/general-mappings/general-mappings.component.ts [hunk 6]
```diff
@@ -47,60 +47,93 @@ export class GeneralMappingsComponent implements OnInit {
...
+      if (that.generalSettings.corporate_credit_card_expenses_object && that.generalSettings.corporate_credit_card_expenses_object === 'BILL') {
+        that.form.controls.defaultVendor.setValidators(Validators.required);
+      }
+
+      if (that.generalSettings.import_projects) {
+        that.form.controls.defaultItem.setValidators(Validators.required);
+      }
+
+      if (that.generalSettings.sync_fyle_to_sage_intacct_payments) {
+        that.form.controls.paymentAccount.setValidators(Validators.required);
```
Comment (ashwin1111) [inline src/app/si/settings/general-mappings/general-mappings.component.ts:118]: Will payment account be used for both reimbursements as well as ap payments? If not, do we need anything else as mandatory for one of the payment type?

File: src/app/si/settings/general-mappings/general-mappings.component.ts [hunk 7]
```diff
@@ -47,60 +47,93 @@ export class GeneralMappingsComponent implements OnInit {
...
+
+      if (that.generalSettings.corporate_credit_card_expenses_object && this.generalSettings.corporate_credit_card_expenses_object === 'CHARGE_CARD_TRANSACTION') {
+        that.form.controls.chargeCard.setValidators(Validators.required);
+      }
+
+      if (that.generalSettings.corporate_credit_card_expenses_object && that.generalSettings.corporate_credit_card_expenses_object === 'BILL') {
+        that.form.controls.defaultVendor.setValidators(Validators.required);
+      }
+
+      if (that.generalSettings.import_projects) {
```
Comment (ashwin1111) [inline src/app/si/settings/general-mappings/general-mappings.component.ts:113]: Since we depreciated this flag, please modify this condition

Comment (ashwin1111) [reply src/app/si/settings/general-mappings/general-mappings.component.ts:113]: Check this across all apps please

**Review by ashwin1111 — CHANGES_REQUESTED** [review]

File: src/app/si/settings/general-mappings/general-mappings.component.ts (hunk 6 above)
Comment (NileshPant1999) [reply src/app/si/settings/general-mappings/general-mappings.component.ts:118]: yes payment account is used for both ap payment as well as reimbursement

Comment (ashwin1111) [reply src/app/si/settings/general-mappings/general-mappings.component.ts:118]: Please include it in the condition above. 
`if (that.generalSettings.sync_fyle_to_sage_intacct_payments || other_column) { }`

Comment (ashwin1111) [reply src/app/si/settings/general-mappings/general-mappings.component.ts:118]: I don't think we use it for reimbursements, please check the whole flow and confirm

Comment (ashwin1111) [reply src/app/si/settings/general-mappings/general-mappings.component.ts:118]: Please add missing validations in api if you add it here

Comment (ashwin1111) [reply src/app/si/settings/general-mappings/general-mappings.component.ts:118]: resolved offline

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**